core/src/main/java/org/apache/accumulo/core/fate/FateExecutor.java

```diff
@@ -306,18 +306,19 @@ public void run() {
             }
           });
         } catch (Exception e) {
-          if (!fate.getKeepRunning().get() || isShutdown()) {
-            log.debug("Expected failure while attempting to find work for fate: either fate is "
-                + "being shutdown and therefore all fate threads are being shutdown or the "
-                + "fate threads assigned to work on {} were invalidated by config changes "
-                + "and are being shutdown", fateOps, e);
-          } else {
-            log.warn("Unexpected failure while attempting to find work for fate", e);
-          }
-
+          log.warn("Unexpected failure while attempting to find work for fate", e);
           workQueue.clear();
         }
       }
+
+      if (!fate.getKeepRunning().get() || isShutdown()) {
```

Comment:
Maybe this if is not needed.  Seems like these should always be true the way the code is structured.  Was wondering if there should be else for this if, looking at the code not sure what the else should be.